### PR TITLE
#2 - Users.FromStorageYaml removed

### PR DESCRIPTION
**Pull request by olenagerasimova**
Part of #2 
I've fully implemented `FakeUsers` and used it in all tests, `Users.FromStorageYaml` is removed.

### CODE REVIEW

File: src/main/java/com/artipie/management/Users.java [hunk 1]
```diff
@@ -347,7 +169,8 @@ public boolean equals(final Object other) {
                 final User user = (User) other;
                 res = Objects.equals(this.uname, user.uname)
                     && Objects.equals(this.mail, user.mail)
-                    && Objects.equals(this.ugroups, user.ugroups);
+                    && this.ugroups.size() == user.ugroups.size()
+                    && this.ugroups.containsAll(user.ugroups);
```
Comment (olegmoz) [inline src/main/java/com/artipie/management/Users.java:173]: @olenagerasimova could you please explain why these changes are needed? They are possibly making `equals` and `hashCode` inconsistent. If this is done for tests, then it should be test-specific code

**Review by olegmoz — CHANGES_REQUESTED** [review]
@olenagerasimova thanks, please see one concern

File: src/main/java/com/artipie/management/Users.java (hunk 1 above)
Comment (olenagerasimova) [reply src/main/java/com/artipie/management/Users.java:173]: @olegmoz groups order does not really matter, that's why I check the size and that one collection contains all elements as other

Comment (olegmoz) [reply src/main/java/com/artipie/management/Users.java:173]: @olenagerasimova the problem is that order might matter in `hashCode` calculation code. I suggest to change `ugroups` type to `Set` and revert code in `equals` method. Current version prone to breaking `equals/hashCode` contract

**Comment by olenagerasimova** [comment]
@olegmoz thanks, fixed

File: src/main/java/com/artipie/management/Users.java [hunk 2]
```diff
@@ -347,7 +170,8 @@ public boolean equals(final Object other) {
...
```
Comment (olegmoz) [inline src/main/java/com/artipie/management/Users.java:174]: @olenagerasimova it seems that you forgot to revert changes in `Users.equals()` method

**Review by olegmoz — CHANGES_REQUESTED** [review]
@olenagerasimova thanks, please see the comment

**Comment by olenagerasimova** [comment]
@olegmoz thanks again, reverted

**Review by olegmoz — APPROVED** [review]
@olenagerasimova great, merging
